### DISCUSSION TITLE
Mypy conflicts with sqlmodel

### DISCUSSION THREAD

**Discussion by hootpy** [fastapi/sqlmodel#606] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/", response_model=list[User])
async def get_users(skip: int = 0, limit: int = 100, db: AsyncSession = Depends(get_session)) -> list[User]: # <== error here
    return await UserCrud.get_users(db, skip, limit)


# Fixed for mypy but show error in pycharm
@router.get("/", response_model=list[User])
async def get_users(skip: int = 0, limit: int = 100, db: AsyncSession = Depends(get_session)) -> User:
    return await UserCrud.get_users(db, skip, limit)

# get_users function
@classmethod
    async def get_users(cls, db: AsyncSession, skip: int = 0, limit: int = 100) -> list[User]:
        query = await db.execute(select(User).offset(skip).limit(limit))
        return query.scalars().all()
```

### Description

- Using with Fastapi, create a crud module
- When using in router, mypy says error: `Return Any  from function declared to return List[Any]`
- Fixed by change return type to `User` but the Pycharm says error type as it not should be like this

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

3.11.3

**Comment by hootpy** [comment]
original code:
![image](https://github.com/tiangolo/sqlmodel/assets/45201848/b15032bc-0cca-46d5-b42b-c3a988ab3cd2)
reference code:
![image](https://github.com/tiangolo/sqlmodel/assets/45201848/7e9529de-79d2-4036-85a7-e9f94ef2bdad)
mypy message:
![image](https://github.com/tiangolo/sqlmodel/assets/45201848/aacdf7ac-378c-456f-a1cb-fc1f9386978a)
fixed for mypy:
![image](https://github.com/tiangolo/sqlmodel/assets/45201848/3560c341-2155-48f1-a666-174d67e090a3)

**Comment by h-wei** [comment]
You can try this:
```
@classmethod
async def get_users(cls, db: AsyncSession, skip: int = 0, limit: int = 100) -> list[User]:
    query = await db.execute(select(User).offset(skip).limit(limit))
    # add the return variable with type hint
    result: list[User] = query.scalars().all()
    return result
```

**Comment by YuriiMotov** [comment]
Closing this as there is not enough info to reproduce the issue locally
